From 0905dd4bbbbbce30753a4c63f29a56b3d667da0c Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 18 Jul 2017 14:59:33 +0800
Subject: [PATCH 2151/5242] MLK-16015-2 phy: mixel-lvds-combo: Add delay to
 wait PHY to be locked

commit  e944130d14da8ed98109e46f5b7b55e3fc84408b from
https://source.codeaurora.org/external/imx/linux-imx.git

It seems that we haven't got SCU ISO bit available to check if PHY is
locked or not after enable, so let's simply delay for a while as a
temporary solution.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-mixel-lvds-combo.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/phy/phy-mixel-lvds-combo.c b/drivers/phy/phy-mixel-lvds-combo.c
index 0005cf2..9d3d469 100644
--- a/drivers/phy/phy-mixel-lvds-combo.c
+++ b/drivers/phy/phy-mixel-lvds-combo.c
@@ -13,6 +13,7 @@
  */
 
 #include <linux/clk.h>
+#include <linux/delay.h>
 #include <linux/io.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
@@ -174,6 +175,8 @@ static int mixel_lvds_combo_phy_power_on(struct phy *phy)
 	phy_csr_write(phy, val, PHY_CTRL);
 	mutex_unlock(&lvds_phy->lock);
 
+	usleep_range(500, 1000);
+
 	return 0;
 }
 
-- 
1.7.9.5

